geozero/src/wkb/wkb_reader.rs

```diff
@@ -70,29 +106,29 @@ pub fn process_wkb_type_geom<R: Read, P: GeomProcessor>(
         WkbDialect::Wkb => process_wkb_geom(raw, processor),
         WkbDialect::Ewkb => process_ewkb_geom(raw, processor),
         WkbDialect::Geopackage => process_gpkg_geom(raw, processor),
+        WkbDialect::SpatiaLite => process_spatialite_geom(raw, processor),
+        WkbDialect::MySQL => process_mysql_geom(raw, processor),
     }
 }
 
 #[derive(Debug)]
 pub(crate) struct WkbInfo {
-    endian: scroll::Endian,
+    endian: Endian,
     base_type: WKBGeometryType,
     has_z: bool,
     has_m: bool,
     #[allow(dead_code)]
     srid: Option<i32>,
     #[allow(dead_code)]
     envelope: Vec<f64>,
+    is_compressed: bool,
 }
 
 /// OGC WKB header.
 pub(crate) fn read_wkb_header<R: Read>(raw: &mut R) -> Result<WkbInfo> {
     let byte_order = raw.ioread::<u8>()?;
-    let endian = if byte_order == WKBByteOrder::Xdr as u8 {
-        scroll::BE
-    } else {
-        scroll::LE
-    };
+    let is_little_endian = byte_order != 0;
```

Comment:
turns out there is a much cleaner solution, but it requires a bit of research.  https://docs.rs/scroll/latest/scroll/ has a lot of docs for that, but essentially instead of using `raw.ioread::<u8>()` you can use `raw.ioread::<WKBByteOrder>`. In order for that to work, you would have to implement `impl IntoCtx<scroll::Endian> for WKBByteOrder {` or something similar (I never used this crate, just saw some docs about cread vs gread vs pread - not sure which is which).  At the end of the day, it should offer a very easy way to just do all the on-the-fly conversions cleanly

Reply:
P.P.S. note that we don't need to fix ALL such issues at the same time, can be done in another PR.  For more docs on this - https://github.com/m4b/scroll#stdio-api